modules/core/src/main/java/org/apache/ignite/configuration/ClientConfiguration.java

```diff
@@ -115,18 +125,35 @@
     /** Reconnect throttling retries. See {@code reconnectThrottlingPeriod}. */
     private int reconnectThrottlingRetries = 3;
 
+    /** Retry limit. */
+    private int retryLimit = 0;
+
     /**
      * @return Host addresses.
      */
     public String[] getAddresses() {
-        return addrs;
+        return Optional.ofNullable(addrFinder)
+            .map(Supplier::get)
+            .orElse(addrs);
     }
 
     /**
      * @param addrs Host addresses.
      */
     public ClientConfiguration setAddresses(String... addrs) {
-        this.addrs = addrs;
+        if (addrs != null) {
```

Comment:
Same here - let's revert and move the logic to `ReliableChannel`.

Reply:
fixed.